### PR TITLE
Renaming hostname to customer_strategy_ref

### PR DESCRIPTION
**Pull request by jsphon**
Renaming `hostname` to `customer_strategy_ref`, which makes intent more explicit.

### CODE REVIEW

File: flumine/config.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 instance_id = None  # instance id (e.g. AWS ec2 instanceId)
 
-hostname = socket.gethostname()[
+customer_strategy_ref = socket.gethostname()[
```
Comment (liampauling) [inline flumine/config.py:8]: Can you add an alias below 

```python
hostname = customer_strategy_ref
```

We can then add a warning before removing it in the history as I know a few have used/changed this.

**Review by liampauling — CHANGES_REQUESTED** [review]

**Comment by ThomasAJ05** [comment]
I'm not sure this is an improvement as currently everywhere I see code like `self.customer_strategy_ref = config.hostname` I know more or less **exactly** what it means. Whereas with `self.customer_strategy_ref = config.customer_strategy_ref` I will (in the future when I have forgotten) I will have to go and check what initialized `config.customer_strategy_ref`.

Or to put it another way, the new code adds an unnecessary layer of possible obfuscation. Looking at '= config.hostname' vs '= config.customer_strategy_ref' I think the former is much clearer.

**Comment by jsphon** [comment]
> I'm not sure this is an improvement as currently everywhere I see code like `self.customer_strategy_ref = config.hostname` I know more or less **exactly** what it means. Whereas with `self.customer_strategy_ref = config.customer_strategy_ref` I will (in the future when I have forgotten) I will have to go and check what initialized `config.customer_strategy_ref`.
> 
> Or to put it another way, the new code adds an unnecessary layer of possible obfuscation. Looking at `= config.hostname` vs `= config.customer_strategy_ref` I think the former is much clearer.

But config.hostname is just being used as the default value for customer_strategy_ref. If someone wants to use a different value, then it's a bit incorrect to call it config.hostname.

**Comment by ThomasAJ05** [comment]
Aha... "If someone wants to use a different value"... yes in that case you are correct.

**Comment by liampauling** [comment]
@ThomasAJ05 that was my reasoning to start with but as @jsphon pointed out it is now only used for this variable so calling it hostname is a bit confusing (regular question on the slack)

**Review by liampauling — APPROVED** [review]

**Pull request merged**